### PR TITLE
Webtest cases written

### PR DESCRIPTION
**Pull request by neetumorwani**

### CODE REVIEW

**Comment by neetumorwani** [comment]
This PR is WIP. Not to be merged.

File: src/Tests/AutologoutAjaxTestCaseTest.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = array('autologout','node','user');
+
+  /**
+   * User with admin rights.
+   */
+  protected $privileged_user;
```
Comment (ajitdev) [inline src/Tests/AutologoutAjaxTestCaseTest.php]: This should be camelCased.

File: src/Tests/AutologoutAjaxTestCaseTest.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    $this->drupalLogin($this->privileged_user);
+
+  }
+
+  /**
+   * Test ajax logout callbacks work as expected.
+   */
+  public function testAutologoutByAjax() {
+
```
Comment (ajitdev) [inline src/Tests/AutologoutAjaxTestCaseTest.php:57]: Remove the extra line.

File: src/Tests/AutologoutAjaxTestCaseTest.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    $autologout_settings = \Drupal::configFactory()->getEditable('autologout.settings');
+    $autologout_settings->set('timeout', 100)
+      ->set('padding', 10)
+      ->save();
+
```
Comment (ajitdev) [inline src/Tests/AutologoutAjaxTestCaseTest.php:62]: Extra line here.

File: src/Tests/AutologoutTestCaseTest.php [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = array('autologout','user','node');
+  /**
+  * User with admin rights.
+  */
+  protected $privileged_user;
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: Class variables should be camelCased.

File: src/Tests/AutologoutTestCaseTest.php [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+  $this->drupalPostForm('admin/config/people/autologout', $edit, t('Save configuration'));
+  $this->assertText(t('The configuration options have been saved.'), t('Unable to save autologout due to out of range role timeout for a role which is not enabled..'));
+  }
+
+  /**
+  * Test enforce logout on admin page settings.
+  */
+  public function testAutlogoutOfAdminPages() {
+
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php:136]: Extra line.

File: src/Tests/AutologoutTestCaseTest.php [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  // Check if user will be kept logged in on admin paths with enforce dsabled.
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('enforce_admin', FALSE)
+      ->save();
+  $this->assertEqual(autologout_autologout_refresh_only(), TRUE, t('Autologout does logout of admin pages without enforce on admin checked.'));
+
+  // Check if user will not be kept logged in on admin paths if enforce enabled.
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('enforce_admin', TRUE)
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: 2 extra spaces here and on the line above.

File: src/Tests/AutologoutTestCaseTest.php [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+      ->save();
+  $this->assertEqual(autologout_autologout_refresh_only(), FALSE, t('Autologout does not logout of admin pages with enforce on admin not checked.'));
+
+  // Set a non admin page path.
+  $_GET['q'] = 'node';
+
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('enforce_admin', FALSE)
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: Extra spaces.

File: src/Tests/AutologoutTestCaseTest.php [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+      ->save();
+  $this->assertEqual(autologout_autologout_refresh_only(), FALSE, t('autologout_autologout_refresh_only() returns FALSE on non admin page when enforce is disabled.'));
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('enforce_admin', TRUE)
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: Extra spaces.

File: src/Tests/AutologoutTestCaseTest.php [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+  $this->assertEqual(autologout_autologout_refresh_only(), FALSE, t('autologout_autologout_refresh_only() returns FALSE on non admin page when enforce is enabled.'));
+  }
+
+  /**
+  * Test a user is logged out and denied access to admin pages.
+  */
+  public function testAutologoutDefaultTimeoutAccessDeniedToAdmin() {
+  // Enforce auto logout of admin pages.
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('enforce_admin', FALSE)
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: Extra spaces.

File: src/Tests/AutologoutTestCaseTest.php [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+  $this->assertText(t('The configuration options have been saved.'), t('Unable to save autologout due to out of range role timeout for a role which is not enabled..'));
+  }
+
+  /**
+  * Test enforce logout on admin page settings.
+  */
+  public function testAutlogoutOfAdminPages() {
+
+  // Set an admin page path.
+  $_GET['q'] = 'admin';
```
Comment (ajitdev) [inline src/Tests/AutologoutTestCaseTest.php]: Indent the code by 4 spaces here. Also check throughout the file for the indentation.

**Comment by ajitdev** [comment]
Thank you @neetumorwani! I mostly see coding standards issues (mainly spacing).

**Comment by neetumorwani** [comment]
Thanks @ajitdev for reviewing. Will fix the coding standards :)

**Pull request merged**